dynatrace/src/main/java/com/dynatrace/opentelemetry/metric/CumulativeToDeltaConverter.java

```diff
@@ -0,0 +1,153 @@
+package com.dynatrace.opentelemetry.metric;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import io.opentelemetry.api.metrics.common.Labels;
+import io.opentelemetry.sdk.metrics.data.DoublePointData;
+import io.opentelemetry.sdk.metrics.data.LongPointData;
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The CumulativeToDeltaConverter provides a way to transform cumulative (total) values to delta
+ * values which can be ingested in Dynatrace.
+ */
+class CumulativeToDeltaConverter {
+  private final Cache<String, CacheValue> cache;
+
+  private static class CacheValue {
+    private final Number number;
+    private final long timeMillis;
+
+    public CacheValue(Number number, long timeMillis) {
+      this.number = number;
+      this.timeMillis = timeMillis;
+    }
+
+    public Number getNumber() {
+      return number;
+    }
+
+    public long getTimeMillis() {
+      return timeMillis;
+    }
+  }
+
+  /**
+   * Create a new converter.
+   *
+   * @param expireAfter The {@link Duration} after which cache entries will expire, calculated from
+   *     the last write operation.
+   */
+  CumulativeToDeltaConverter(Duration expireAfter) {
+    this.cache = CacheBuilder.newBuilder().expireAfterWrite(expireAfter).build();
+  }
+
+  private static String createIdentifier(String name, Labels labels, String type) {
+    return String.format("%s\u001d%s\u001d%s", name, getSortedLabelsString(labels), type);
```

Comment:
```suggestion
    // (\u001d = ASCII group separator)
    return String.format("%s\u001d%s\u001d%s", name, getSortedLabelsString(labels), type);
```
Since that's something one usually never ever encounters in the wild and thus might be confused :)